polygon/p2p/message_listener.go

```diff
@@ -69,6 +74,14 @@ func (ml *messageListener) Stop() {
 	ml.peerEventObservers = map[MessageObserver[*sentry.PeerEvent]]struct{}{}
 }
 
+func (ml *messageListener) RegisterNewBlockObserver(observer MessageObserver[*sentry.InboundMessage]) {
+	ml.registerInboundMessageObserver(observer, sentry.MessageId_NEW_BLOCK_66)
```

Comment:
you need to add a background loop for this message id - check `messageListener.Start`